### ISSUE TITLE
Errata of the documentation of SPH Package 

### ISSUE THREAD

**Issue by Jasmine969** [lammps/lammps#4363]
**Summary**

There are some typos in the [SPH documentation](https://docs.lammps.org/PDF/SPH_LAMMPS_userguide.pdf). In case users new to LAMMPS-SPH may be confused and misled, the errata are listed below. Further contributions are welcome.

**Details**

## Mistakes of the plus/minus sign

1. In Eq. (8), the sign before $-P\cdot\nabla\rho/\rho^2$ should be a minus sign instead of a plus sign, which can be easily deduced from Eq. (7).
![image](https://github.com/user-attachments/assets/8e4ebd0b-f018-48d2-b723-b085894f9375)

2. In Eqs. (15-16), the plus/minus signs are all reversed on the right-hand side. The dot operators are also omitted in Eqs. (14-16). The correct formulae are
$$\nabla\cdot(\mathbf{v} \rho)=\rho \nabla\cdot \mathbf{v}+\mathbf{v}\cdot \nabla \rho,$$
$$\dfrac{\mathrm{d} \rho}{\mathrm{d} t}=\nabla\cdot(\mathbf{v} \rho)-\mathbf{v}\cdot \nabla \rho,$$
and 
$$\dfrac{\mathrm{d} \rho_i}{\mathrm{d} t}=\sum_j m_j \mathbf{v}_j \nabla_j W_{ij}-\mathbf{v}_i \sum_j m_j \nabla_j W_{ij}=-\sum_j m_j \mathbf{v}_{ij} \nabla_j W_{ij}$$
![](https://fengimages-1310812903.cos.ap-shanghai.myqcloud.com/20241028191849.png)
This can be easily deduced from Eq. (1, 15).

## Deprecated and updated features
1. All `meso` should be updated to `sph`. For example, `atom_style meso` -> `atom_style sph`.
2. All `meso_` should be updated to `sph/`. For example, `set style ID meso_rho d` -> `set style ID sph/rho d`.

## Suggested revisions
The kernel gradient $\nabla_j W_{ij}$ in all relevant equations is preferred to be changed to $\nabla_i W_{ij}$ to represent the kernel gradient taken with respect to the coordinates of particle $i$, which is consistent with Monaghan's notation $i$ (Monaghan, 1994). The definition should thus be $\nabla_i W_{i j}=\frac{\mathbf{r}_{i j}}{\left\Vert\mathbf{r}_{i j}\right\Vert} \frac{\mathrm{d} W_{i j}}{\mathrm{d} r_{i j}}$.

## Reference
Monaghan JJ. Simulating free surface flows with SPH. Journal of Computational Physics. 1994;110(2):399-406. doi:[10.1006/jcph.1994.1034](https://doi.org/10.1006/jcph.1994.1034)

**Comment by akohlmey** [comment]
As already discussed in https://matsci.org/t/the-lammps-sph-users-guide-has-to-be-updated/54320/
Without access to the LaTeX sources of the mentioned PDF file, there is nothing that can be done from the side of the LAMMPS developers.

@Jasmine969 @sjplimp Have you since been able to contact the contributing author of the package?

@jtclemm You mentioned that the RHEO package can replace SPH. For that we would need to provide a "transition guide" that shows people how to convert inputs from one package to the other. We could start by trying to convert the examples in the `examples/PACKAGES/sph` sub-folders. That would also provide some insight into what would still be missing for a complete replacement that would allow us to declare SPH obsolete and eventually remove it as unsupported.

Unlike some of the other "exotic" package in LAMMPS, we regularly see people posting questions about SPH simulations, so we have to put in some effort to have a smooth and easy transition.

**Comment by Jasmine969** [comment]
> [@Jasmine969](https://github.com/Jasmine969) [@sjplimp](https://github.com/sjplimp) Have you since been able to contact the contributing author of the package?


Not yet. I emailed the contributing author but haven't received a reply.

**Comment by akohlmey** [comment]
> Not yet. I emailed the contributing author but haven't received a reply.

Sometimes it needs a bit more than that. E.g. you first have to research if that email address is still the correct one. In academia people keep changing jobs and not always will emails to the old address be forwarded to the next. I found that Google scholar or Orcid can be helpful to determine the current affiliation of a researcher and then a websearch usually yields a corresponding email address. The SPH code has not seen any updates from the original author in a very long time, so the embedded email address(es) are not likely to be current.

**Comment by Jasmine969** [comment]
> embedded email address(es) are not likely to be current.

Thanks for your reminder. I found a new email address from https://ggalu.github.io/02_staff.html. I will resend an email to him.

**Comment by jtclemm** [comment]
> [@jtclemm](https://github.com/jtclemm) You mentioned that the RHEO package can replace SPH. For that we would need to provide a "transition guide" that shows people how to convert inputs from one package to the other. We could start by trying to convert the examples in the `examples/PACKAGES/sph` sub-folders. That would also provide some insight into what would still be missing for a complete replacement that would allow us to declare SPH obsolete and eventually remove it as unsupported.
> 
> Unlike some of the other "exotic" package in LAMMPS, we regularly see people posting questions about SPH simulations, so we have to put in some effort to have a smooth and easy transition.

Agreed on a transition guide, but I think it's too early. The SPH package implements a different flavor of SPH than RHEO which generally is a trade-off between ease-of-use and accuracy. So I wouldn't want to shunt users to a different package unless there are options to toggle between the different numerical formulations. Something on my to do list.
